### ISSUE TITLE
Feature request: Transform Processor support in ADOT Collector

### ISSUE THREAD

**Issue by bpholt** [aws-observability/aws-otel-collector#1869]
tl;dr: it would be great if the ADOT collector distribution supported the `transform` processor so users can easily replace illegal characters in span attributes.

We route OTel data to X-Ray, which limits the legal characters in segment names; in particular, `!` characters are not allowed. Unfortunately, our telemetry includes the `peer.service` span attribute with values that look like `protocol!service`, and [if `peer.service` is present, it's used as the segment name](https://aws-otel.github.io/docs/getting-started/x-ray#configuring-the-aws-x-ray-exporter).

If the collector supported the `transform` processor, we could apply the following transformation:

```yaml
processors:
  transform/replace-exclamation-with-colon:
    trace_statements:
    - context: span
      statements:
      - replace_pattern(attributes["peer.service"], "!", ":")
```

Without `transform`, we have a workaround:

```yaml
processors:
  attributes:
    actions:
      - key: peer.service
        action: extract
        pattern: ^(.+)!(?P<peerservice>.*)$    # AFAICT using "peer.service" would be an illegal capture group name
      - key: peer.service
        action: update
        from_attribute: peerservice
      - key: peerservice
        action: delete
```

but this is three steps instead of one, and I'm not confident it will hold up as we scale up the number of spans we're processing.

**Comment by matthewgrossman** [comment]
Subscribed and commenting with my usecase:

We want to translate/map a portion of `Resource` attributes over to `Span` attributes. Both the `ResourceProcessor` and `AttributesProcessor` work within their respective domains.

Some advice on the CNCF slack advised us to use the `TransformProcessor`, but we can't do that because we're running the adotcollector: [CNCF slack](https://cloud-native.slack.com/archives/C01N6P7KR6W/p1675800486128869)

**Comment by bpholt** [comment]
@mhausenblas do you have any update on this request?

**Comment by ademartini-czi** [comment]
We would also like an update on this.  Is there some workaround for getting the resource attributes into labels?

**Comment by RafaelGBAccenture** [comment]
This feature is essential for efficiently processing message content towards CloudWatch. Having the transform processor would enable us to parse and filter logs effectively, allowing for better data management and more accurate insights. This capability is critical for creating efficient log filters and improving overall system observability. Thank you for considering this request.
